_posts/2016-06-11-drupal-7-to-drupal-8-migration-diary-part-2-config-installer.md

```diff
@@ -0,0 +1,59 @@
+* Push committed configuration to GitHub, make pull request.
+* CircleCI then installs with the Standard profile, imports configuration from `sites/default/config`, runs tests.
+
+But there is a problem with this workflow. When attempting to do the import of pre-existing configuration (with `drush config-import`) Drupal will complain with a message that includes:
+
+```
+The import failed due for the following reasons:                         
+Site UUID in source storage does not match the target storage.
+```
+
+As far as Drupal core is concerned, `sites/default/config` is meant for configuration from the same site (or at least different copies of the same site). Doing a fresh install of Drupal core means that the pre-existing config in `sites/default/config` is seen as foreign.
+
+## Enter Configuration Installer
+
+[Configuration Installer](https://www.drupal.org/project/config_installer) allows me to use the workflow that I want. (Thanks to [Alex Pott](https://www.drupal.org/u/alexpott) for writing it!) Drupal bases new installations on installation profiles. In Drupal 8, each install profile is expected to provide the configuration it needs in the form of `.yml` files. For most every install profile, those `.yml` files live inside the directory of the install profile (see the directory structure of core's [Standard](http://cgit.drupalcode.org/drupal/tree/core/profiles/standard?h=8.1.2) and [Minimal](http://cgit.drupalcode.org/drupal/tree/core/profiles/minimal?h=8.1.2) install profiles). Configuration Installer is a way to say "my install profile configuration is actually over in `sites/default/config`." It's a workaround that works well. [Here's a pull request that updates the permissions for the Content Administrator role's configuration and changes some Behat tests accordingly](https://github.com/stevector/nerdologues-d8/pull/67/files).
+
+## How are other people working without Configuration Installer?
+
+I would not need Configuration Installer if I established a canonical Drupal 8 database. With a canonical Drupal 8 database, CircleCI would would grab a copy of the D8 database and import configuration changes rather than installing fresh which creates a new database. Once the Drupal 8 site is live, that will be the workflow. But I don't want to go that workflow yet. A Drupal site has three main parts:
+
+- Version-controlled code.
+- Files ignored from version control (mostly uploaded images/documents).
+- A database.
+
+The database is easiest element of the three to mess up and the hardest to fix after a mistaken change. That's why Drupal 8 introduced new configuration management tools that move focus from the database to version controlled code. *Configuration does still live in the database* as far as a runtime Drupal site is concerned but configuration is so cleanly imported and exported to code now that mental focus goes to the `.yml` files.
```

Comment:
`Of the three, The database is easiest element to mess up and the hardest to fix after a mistaken change.`